core/src/test/java/org/infinispan/api/AsyncAPITest.java

```diff
@@ -91,6 +97,26 @@ public void testPutAsync() throws Exception {
       assertEquals("v2", c.get("k"));
    }
 
+   public void testPutAsyncEntry() throws Exception {
+      Metadata metadata = new EmbeddedMetadata.Builder().version(new NumericVersion(1)).build();
```

Comment:
Metadata checks are good now, however can we have some diversity in the assertions with the versions and expiration? We are only ever inserting version 1 and no expiration and only asserting that. Having a different version and some different expiration times would be greatly appreciated to have some confidence it is working properly.

Reply:
The tests are using a metadata with lifespan and max idle and updating the values between operations.